### ISSUE TITLE
Wine crashes in LinearAlgebra/triangular

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#43583]
Looks like this crash is in unwind:
```
#0  0x0000000170057a0f in RtlVirtualUnwind (type=type@entry=0, base=base@entry=1723955056, pc=1723955349, function=0x7f233c6da640, context=<optimized out>,
    context@entry=0xc2ddc0, data=0xc2d818, data@entry=0xc2d888, frame_ret=frame_ret@entry=0xc2d820, ctx_ptr=<optimized out>, ctx_ptr@entry=0x0)
    at ../dlls/ntdll/signal_x86_64.c:863
#1  0x0000000063daf182 in jl_unw_step (sp=<synthetic pointer>, ip=<synthetic pointer>, from_signal_handler=0, cursor=0xc2ddc0) at /home/keno/julia-mywin/src/stackwalk.c:521
#2  jl_unw_stepn (cursor=0xc2ddc0, cursor@entry=0xc2ddf0, bt_data=0x0, bt_data@entry=0xd490030, bt_size=0x1fa00010000f, bt_size@entry=0xc2d918, sp=0x33, sp@entry=0x0,
    maxsize=maxsize@entry=80000, skip=0, skip@entry=3, ppgcstack=0xc2d8e0, ppgcstack@entry=0xc2d910, from_signal_handler=from_signal_handler@entry=0)
    at /home/keno/julia-mywin/src/stackwalk.c:99
#3  0x0000000063daf6d3 in rec_backtrace (bt_data=0xd490030, maxsize=maxsize@entry=80000, skip=skip@entry=2) at /home/keno/julia-mywin/src/stackwalk.c:222
#4  0x0000000063d7cb96 in record_backtrace (ptls=0x1c4770, skip=skip@entry=1) at /home/keno/julia-mywin/src/task.c:344
#5  0x0000000063d7d230 in ijl_throw (e=0x7f23542c40a0) at /home/keno/julia-mywin/src/task.c:649
#6  0x0000000063d445c8 in ijl_error (str=str@entry=0x63ee8f30 <system_image_path+2288> "NamedTuple field type must be a tuple type") at /home/keno/julia-mywin/src/rtutils.c:41
#7  0x0000000063d4c45d in inst_datatype_inner (dt=0x7f232c5bd610, p=<optimized out>, p@entry=0x0, iparams=iparams@entry=0x12585210 <jl_system_image_data+422416>,
    ntp=ntp@entry=2, stack=0xc2e480, stack@entry=0x0, env=env@entry=0xc2e540) at /home/keno/julia-mywin/src/jltypes.c:1509
#8  0x0000000063d4c58b in inst_datatype_env (dt=<optimized out>, iparams=iparams@entry=0x12585210 <jl_system_image_data+422416>, ntp=ntp@entry=2, env=env@entry=0xc2e540,
    c=c@entry=2, stack=0x0, p=0x0) at /home/keno/julia-mywin/src/jltypes.c:909
#9  0x0000000063d4c567 in inst_datatype_env (dt=<optimized out>, iparams=iparams@entry=0x12585210 <jl_system_image_data+422416>, ntp=ntp@entry=2, env=env@entry=0xc2e5a0,
    c=c@entry=1, stack=0x0, p=0x0) at /home/keno/julia-mywin/src/jltypes.c:913
#10 0x0000000063d4c567 in inst_datatype_env (dt=dt@entry=0x126be660 <jl_system_image_data+1705568>, iparams=iparams@entry=0x12585210 <jl_system_image_data+422416>,
    ntp=ntp@entry=2, env=env@entry=0x0, c=c@entry=0, stack=0x0, p=0x0) at /home/keno/julia-mywin/src/jltypes.c:913
#11 0x0000000063d4d62c in ijl_apply_type (tc=<optimized out>, params=0x12585210 <jl_system_image_data+422416>, params@entry=0x0, n=2, n@entry=0)
    at /home/keno/julia-mywin/src/jltypes.c:929
#12 0x0000000063dc92ab in intersect (x=0xc2f730, y=y@entry=0x17086b30 <jl_system_image_data+79072048>, e=e@entry=0xc333c0, param=param@entry=12776924)
    at /home/keno/julia-mywin/src/subtype.c:3138
#13 0x0000000063dd0f7f in intersect_unionall_ (t=t@entry=0x17086b30 <jl_system_image_data+79072048>, u=<optimized out>, u@entry=0x13cd9700 <jl_system_image_data+24884992>,
    e=e@entry=0xc333c0, R=R@entry=0 '\000', param=2, param@entry=374626448, vb=vb@entry=0xc2f5e0) at /home/keno/julia-mywin/src/subtype.c:2578
#14 0x0000000063dd11ee in intersect_unionall (t=t@entry=0x17086b30 <jl_system_image_data+79072048>, u=0x13cd9700 <jl_system_image_data+24884992>, u@entry=0xc306dc,
    e=e@entry=0xc333c0, R=R@entry=0 '\000', param=<optimized out>, param@entry=2) at /home/keno/julia-mywin/src/subtype.c:2623
#15 0x0000000063dc888a in intersect (x=0xc306dc, x@entry=0x13cd9700 <jl_system_image_data+24884992>, y=y@entry=0x17086b30 <jl_system_image_data+79072048>, e=e@entry=0xc333c0,
    param=param@entry=2) at /home/keno/julia-mywin/src/subtype.c:3106
#16 0x0000000063dce440 in intersect_invariant (x=0x13cd9700 <jl_system_image_data+24884992>, y=0x17086b30 <jl_system_image_data+79072048>, e=e@entry=0xc333c0)
    at /home/keno/julia-mywin/src/subtype.c:2869
#17 0x0000000063dc9133 in intersect (x=0x126be6c0 <jl_system_image_data+1705664>, x@entry=0x13cd9690 <jl_system_image_data+24884880>,
    y=y@entry=0x17086ac0 <jl_system_image_data+79071936>, e=e@entry=0xc333c0, param=param@entry=0) at /home/keno/julia-mywin/src/subtype.c:3130
#18 0x0000000063dc94be in intersect_tuple (xd=0xc32460, param=12789096, e=0xc333c0, yd=0x63dc8a36 <intersect+2806>) at /home/keno/julia-mywin/src/subtype.c:2770
#19 intersect (x=0xc32460, x@entry=0x7f236f59df50, y=0x63dc8a36 <intersect+2806>, e=e@entry=0xc333c0, param=12789096, param@entry=307374432)
    at /home/keno/julia-mywin/src/subtype.c:3124
#20 0x0000000063dd104a in intersect_unionall_ (t=t@entry=0x7f236f59df50, u=<optimized out>, u@entry=0x17086a10 <jl_system_image_data+79071760>, e=e@entry=0xc333c0,
    R=R@entry=1 '\001', param=param@entry=0, vb=vb@entry=0xc32310) at /home/keno/julia-mywin/src/subtype.c:2574
#21 0x0000000063dd11ee in intersect_unionall (t=0x7f236f59df50, t@entry=0xc2f5e0, u=u@entry=0x17086a10 <jl_system_image_data+79071760>, e=e@entry=0xc333c0, R=R@entry=1 '\001',
    param=<optimized out>, param@entry=0) at /home/keno/julia-mywin/src/subtype.c:2623
#22 0x0000000063dc8a36 in intersect (x=0xc2f5e0, x@entry=0xc33348, y=0x17086a10 <jl_system_image_data+79071760>, y@entry=0x1, e=e@entry=0xc333c0, param=param@entry=0)
    at /home/keno/julia-mywin/src/subtype.c:3109
#23 0x0000000063dc9e3b in intersect_all (x=0xc33348, x@entry=0x7f236f59df50, y=0x1, y@entry=0x17086a10 <jl_system_image_data+79071760>, e=0xc333c0, e@entry=0xc33440)
    at /home/keno/julia-mywin/src/subtype.c:3175
#24 0x0000000063dd2536 in jl_type_intersection_env_s (a=0x7f236f59df50, b=0x17086a10 <jl_system_image_data+79071760>, penv=0x6a713450 <jl_datatype_type>,
    issubty=issubty@entry=0xc33af8) at /home/keno/julia-mywin/src/subtype.c:3417
#25 0x0000000063d5cf8d in jl_typemap_intersection_node_visitor (ml=0x17038ce0 <jl_system_image_data+78752992>, ml@entry=0x17038c40 <jl_system_image_data+78752832>,
    closure=closure@entry=0xc33ad0) at /home/keno/julia-mywin/src/typemap.c:459
#26 0x0000000063d5ede8 in jl_typemap_intersection_visitor (map=<optimized out>, offs=0, offs@entry=1, closure=closure@entry=0xc33ad0) at /home/keno/julia-mywin/src/typemap.c:626
#27 0x0000000063d5edc7 in jl_typemap_intersection_visitor (map=<optimized out>, map@entry=0x125232c0 <jl_system_image_data+21184>, offs=307374416, offs@entry=0,
    closure=0xc33ad0, closure@entry=0xc33b90) at /home/keno/julia-mywin/src/typemap.c:571
#28 0x0000000063d558d3 in ml_matches (mt=0xc33c80, mt@entry=0x12523260 <jl_system_image_data+21088>, type=type@entry=0x7f236f59df50, lim=0, lim@entry=3,
    include_ambiguous=2015945992, include_ambiguous@entry=0, intersections=<optimized out>, intersections@entry=1, world=<optimized out>, world@entry=12795136,
    cache_result=<optimized out>, cache_result@entry=1, min_valid=<optimized out>, min_valid@entry=0x0, max_valid=<optimized out>, max_valid@entry=0xffffffffffffffff,
    ambig=<optimized out>, ambig@entry=0xc33c78, offs=0) at /home/keno/julia-mywin/src/gf.c:2819
#29 0x0000000063d58d6c in ijl_matching_methods (types=0x7f236f59df50, mt=0x12523260 <jl_system_image_data+21088>, lim=3, include_ambiguous=0, world=31946, min_valid=0xc33c90,
    max_valid=0xc33c80, ambig=0xc33c78) at /home/keno/julia-mywin/src/gf.c:1898
#30 0x00000000121ce041 in _methods_by_ftype () at reflection.jl:931
#31 #findall#252 () at compiler/methodtable.jl:68
#32 findall##kw () at compiler/methodtable.jl:65
#33 julia_#255_17258 () at compiler/methodtable.jl:97
#34 0x0000000300000001 in ?? ()
#35 0x0000000000000000 in ?? ()
```

**Comment by Keno** [comment]
I'm guessing this is our old friend the section offset truncation issue again.

**Comment by Keno** [comment]
@vtjnash points out that the UnwindData is supposed to be in the text section. I'll look at why that's not happening.

**Comment by JeffBezanson** [comment]
Bump

**Comment by Keno** [comment]
Fixed by #44134.
